lib/class-wp-rest-menus-controller.php

```diff
@@ -189,6 +189,10 @@ public function prepare_item_for_response( $term, $request ) {
 		$fields = $this->get_fields_for_response( $request );
 		$data   = $response->get_data();
 
+		if ( in_array( 'locations', $fields, true ) ) {
+			$data['locations'] = $this->get_menu_locations( $nav_menu->term_id );
+		}
```

Comment:
Why is this needed? You can get menu locations from embedding request?

`?_embed=menu-location` at the end of the request.

Reply:
Also, you can just add locations like this. You would need to schema.

Reply:
<img width="828" alt="Screenshot 2021-05-10 at 12 17 49" src="https://user-images.githubusercontent.com/237508/117651410-d2311500-b189-11eb-8935-42feeb4470c5.png">

Reply:
Primarily because of consistency when using with entities on the JS side. You can set locations using `editEntityRecord`, but the `getEntityRecord` doesn't return them.

The `locations` is also defined in the schema but never returned.

> ?_embed=menu-location at the end of the request.

I just tested this, and it doesn't embed menu locations for some reason.

Reply:
In a way, this is similar to how a post response has an array of category/tag IDs.